### ISSUE TITLE
Default button causing `plain` on TextInput, Select, etc. doesn't allow keyboard visual indication of active suggestion/option

### ISSUE THREAD

**Issue by taysea** [grommet/grommet#4142]
### Expected Behavior

When default button is defined in theme and user uses keyboard to move through textinput suggestions, select options, etc., the hover styling should still be applied.

### Actual Behavior

No styling is being applied because the button is marked as `plain`, so there's no indication of which suggestion or select option is active. This only is an issue when default button is defined on theme.

These changes are rooted in this PR: https://github.com/grommet/grommet/pull/4131
However, this PR is solving some issues we were facing such as unwanted padding, bold font, etc. but it is the reason that the keyboard styling is not visible.

### URL, screen shot, or Codepen exhibiting the issue

Current implementation with grommet-theme-hpe-next. I am using keyboard to navigate but can't get the visual indication of which option i'm on:
![plain](https://user-images.githubusercontent.com/12522275/83687719-06c34680-a5a1-11ea-9b1e-77ce0390b59b.gif)

With plain removed (however, this causes unwanted implications of extra padding, bold font, etc.):
![notplain](https://user-images.githubusercontent.com/12522275/83687772-1a6ead00-a5a1-11ea-965f-1549b17ad9a1.gif)

### Steps to Reproduce

See the above gifs

### Your Environment

- Grommet version: stable
- Browser Name and version: chrome
- Operating System and version (desktop or mobile): macos

**Comment by ericsoderberghp** [comment]
I think the crux of this issue is that Menu, Select, TextInput, etc. all use Buttons within the drop contents but explicitly set `pad="small"`, which means that the theme cannot control the padding of those elements.

Option 1

If we change these instances of Button to not be plain, they will have the correct padding and coloring. Unfortunately though, the font weight isn't controllable here. The HPE theme wants the Menu control Button to match the styling of other Buttons. But, it wants the items in the Menu drop to have regular weight text.

Option 2

If we leave these internal Buttons plain when we have a `theme.button.default`, we could explicitly set `pad={theme.button.default.padding}` on these internal Buttons. This would give us the right font weight. But, it feels like it is too opinionated within grommet to hard code this way.

I am leaning towards Option 1, adding `theme.menu.items` with a structure similar to `theme.button.default`, and then also adding font styling to that structure. That would allow the necessary alignment across uses of Button while adding the flexibility for variations as the theme chooses.

**Comment by taysea** [comment]
I agree that option 2 feels like it could be a little too specific to our use case. Extending menu theme seems aligned with what is already established for `theme.select.options`, so it seems like a better approach.

**Comment by ericsoderberghp** [comment]
An update ...

I'm thinking of adding an internal `kind="option"` that Menu, Select, TextArea, etc. would use that would allow styling them all consistently within the theme. The styling would be in `theme.button.option` as opposed to replicating the styling under `theme.menu`, `theme.select`, etc.

**Comment by taysea** [comment]
I would be fine with that as long as the individual theming that can be done on each of those components persists. 

The only thing is that this feels a little hidden from a user's perspective, so thinking about how to document it clearly will be important. Since I'm not technically building Menu with Button, I'm just saying `<Menu items={[...]} />`, revealing what is affected by `kind="option"` will be important.

**Comment by ericsoderberghp** [comment]
agreed
